platforms/jvm/language-java/src/integTest/groovy/org/gradle/api/tasks/compile/JavaCompileProblemsIntegrationTest.groovy

```diff
@@ -304,6 +304,9 @@ class JavaCompileProblemsIntegrationTest extends AbstractIntegrationSpec impleme
     @Issue("https://github.com/gradle/gradle/pull/29141")
     @Requires(IntegTestPreconditions.Java8HomeAvailable)
     def "compiler warnings causes failure in problem mapping under JDK8"() {
+        // We are interested in how the reporting mechanism behaves when the compiler breaks, and not in the problems themselves
+        disableProblemsApiCheck()
```

Comment:
Because the key assumption here is that we have a message arriving at the console. Do you think we - and later on everybody - to assert all problems in tests, even if the focus is not on the problems themselves?

Reply:
Yesterday I was thinking about this, and I changed my mind: in here, we _should_ track the problems, as this is a feature which can change the layout of the problems.

In addition, realized that I've committed a mistake: I should add an unhappy path with >JDK8

Reply:
Added a new test covering all other >8 JDKs